client/src/IgocExplorer/explorer_view.js

```diff
@@ -179,7 +190,13 @@ const get_non_col_content_func = createSelector(
 class ExplorerForIgoc extends React.Component {
   constructor(){
     super();
-    this.state = { _query: "" };
+    this.state = {
+      _query: "",
+
+      // note use_legal_titles exists totally outside the redux here. No need for it to be redux-state,
+      // and not worth all that added boiler plate to get it in there
+      use_legal_titles: false,
+    };
     this.debounced_set_query = _.debounce(this.debounced_set_query, 500);
```

Comment:
Sorta more just a personal question for clarification, but why is this constructor debounced 500ms? Shouldn't it only run whenever the `ExplorerForIgoc` is called, which seems to be only once when `ExplorerContainer` is run

Reply:
The constructor of a React component is only called once right before it first mounts, yeah. Nothing is being debounced within the constructor though, `_.debounce` takes a function and a debounce time and returns a new, debouncing, version of that function. What's happening here is that a new debouncing function is being created, it's not called until later.

I tried to clean the code up a bit anyway, still a mess because there's a weird thing going on with the loading state in this component...

Reply:
Ohh okay. Is that the `.debounced_set_query` part that's the new function?

Reply:
Yeah, although it was a bit messy because the class had a pre-defined function named `debounced_set_query` that was not initially debounced but was being overwritten with a debounced version in the constructor. I cleaned that mess up.

Reply:
Ohhh makes a lot more sense. I don't think that's messy per-se, I just didn't get the whole debouncing at first